lib/ansible/module_utils/keycloak.py

```diff
@@ -66,6 +72,41 @@ def camel(words):
     return words.split('_')[0] + ''.join(x.capitalize() or '_' for x in words.split('_')[1:])
 
 
+def check_role_representation(rep):
+    """
+    This checks whether a given dict is a valid role representation and provides error messages if not.
+
+    :param rep: role representation / specification to be checked
+    :return: Boolean for success, and message if an error occurred as a string.
+    """
+
+    valid_role_params = ['clientRole', 'composite', 'composites', 'containerId', 'description',
+                         'id', 'name', 'scopeParamRequired']
+
+    if not isinstance(rep, dict):
+        return False, 'Roles must be defined as dicts, "%s" is not one.' % rep
+    for k in rep.keys():
```

Comment:
Thanks for this; I'll see about getting it in on the weekend/Monday (time permitting). 

I have tried to get the type-checking done through the argumentspec, though in this case that is not possible. Role representations are well-defined in Keycloak, so at least it was possible here.

I had not thought about adding more logical checks as yet, no, though in a future keycloak_roles module it might be useful (this was also the thinking in putting this in the modutils file instead of with the scopes module).